### PR TITLE
Initial Implementation of Authentication

### PR DESCRIPTION
**Pull request by jainr5668**
This PR brings in the changes required for the authentication to work in backend, supporting few pre-defined role based authentication and a public method to create custom RBAC.

### CODE REVIEW

**Comment by jainr5668** [comment]
@rohitjain25 Please review the PR.

File: test/tests/backend/endpoints/authentication/test_login.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    )
+
+    framework.log.step("Call /endpoints/auth/login with verb OPTIONS")
+    response = framework.requests.request(method="options", url=url, headers=headers)
+    assert Status_Code.METHOD_NOT_ALLOWED == response.status_code, framework.log.error(
+        f"Expected status code is {Status_Code.METHOD_NOT_ALLOWED} but actual is {response.status_Code}"
+    )
+
+
+def test_auth_login_invalid_email(framework):
```
Comment (jainr5668) [inline test/tests/backend/endpoints/authentication/test_login.py:67]: Working on this one.

File: try.py
Comment (jainr5668) [inline try.py:1]: Pushed by mistake, will remove the file in next commit.

File: backend_2/fw/database/__init__.py [hunk 2]
```diff
@@ -1,14 +1,120 @@
...
+            self.__db.commit()
+            data = {}
+            for key, value in zip(self.table_columns(table_name), values):
+                data[key] = value
+            return data
+        except Exception as e:
+            return None
+
+
+class _MongoDB_CLient:
```
Comment (jainr5668) [inline backend_2/fw/database/__init__.py:98]: This class will be updated in upcoming story [LENS-11(Working on the MongoDb class for the database module)](https://developments.atlassian.net/browse/LENS-11)

**Review by rohitjain25 — APPROVED** [review]

**Pull request merged**